test/test_synthetic.py

```diff
@@ -0,0 +1,407 @@
+        }
+        return dct
+
+
+def dl_real(reg='LA', mod='GMAO'):
+    """ Download the real weather model to overwrite
+
+    This 'golden dataset' shouldnt be changed
+    """
+    SAobj = StudyArea(reg, mod)
+    dct_cfg = SAobj.make_config_dict()
+    # set the real weather model path and download only
+    dct_cfg['runtime_group']['weather_model_directory'] = \
+            op.dirname(SAobj.path_wm_real)
+    dct_cfg['download_only'] = True
+
+    cfg = update_yaml(dct_cfg)
+
+    ## run raider to download the real weather model
+    cmd  = f'raider.py {cfg}'
+
+    proc = subprocess.run(cmd.split(), stdout=subprocess.PIPE, universal_newlines=True)
+    assert np.isclose(proc.returncode, 0)
+    return
+
+
+def test_hydrostatic_eq(reg='Fort', mod='GMAO'):
+    """ Test the hydrostatic eqn by setting t=p and comparing to ray
+
+    The ray length is computed here (length_of_ray; m), and in raider (m * K/Pa)
+    ray length computed here is then scaled by constant K/Pa
+    ray length from raider is unscaled from parts per million
+    We check they are both large enough for meaningful numerical comparison (>1)
+    We then compute residual and normalize it we theoretical ray length
+    We ensure that normalized residual is not significantly different from 0
+        significantly different = 7 decimal places
+    """
+
+    ## setup the config files
+    SAobj = StudyArea(reg, mod)
+    dct_cfg      = SAobj.make_config_dict()
+    wm_dir_synth = op.dirname(SAobj.path_wm_real).replace('real', 'synth')
+    dct_cfg['runtime_group']['weather_model_directory'] = wm_dir_synth
+    dct_cfg['download_only'] = False
+
+    ## update the weather model; t = p for hydrostatic
+    path_synth = update_model(SAobj.path_wm_real, 'hydro', wm_dir_synth)
+
+    cfg = update_yaml(dct_cfg)
+
+    ## run raider with the synthetic model
+    cmd  = f'raider.py {cfg}'
+    proc = subprocess.run(cmd.split(), stdout=subprocess.PIPE, universal_newlines=True)
+    assert np.isclose(proc.returncode, 0)
+
+    # get the just created synthetic delays
+    ds = xr.open_dataset(
+        f'{SAobj.wd}/{SAobj.WM}_tropo_{SAobj.dts.replace("_", "")}_ray.nc')
+    da = ds['hydro']
+    ds.close()
+    del ds
+
+    # now build the rays at the unbuffered wm nodes
+    targ_xyz = [da.x.data, da.y.data, da.z.data]
+    ray_length = length_of_ray(targ_xyz, SAobj.wmObj._zlevels, SAobj.los)
+
+    # scale by constant (units K/Pa) to match raider (m K / Pa)
+    ray_data  = ray_length * SAobj.wmObj._k1
+
+    # actual raider data
+    # undo scaling of ppm;  units are  meters * K/Pa
+    raid_data = da.data * 1e6
+
+    assert np.all(np.abs(ray_data) > 1)
+    assert np.all(np.abs(raid_data) > 1)
+
+    # normalize with the theoretical data and compare difference with 0
+    resid     = (ray_data - raid_data) / ray_data
+    np.testing.assert_almost_equal(0, resid, decimal=7)
+
+    da.close()
+    del da
+    return
+
+
+def test_wet_eq1(reg='Fort', mod='GMAO'):
```

Comment:
I believe you don't need to repeate your inputs if you use parametrization in pytest: https://www.tutorialspoint.com/pytest/pytest_parameterizing_tests.htm